### PR TITLE
feat(validation): add closure driven rule

### PR DESCRIPTION
**Pull request by Phil-Venter**
> I think we should also support closure-based rules. I've wanted that a few times, once when writing the release script, and another time when working on the publish command. Writing a class is a bit cumbersome in that situation.
> Enzo

### CODE REVIEW

**Comment by Phil-Venter** [comment]
RE-opened because of https://github.com/tempestphp/tempest-framework/pull/570#issuecomment-2408387865

File: src/Tempest/Validation/src/Rules/Custom.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+declare(strict_types=1);
+
+namespace Tempest\Validation\Rules;
+
+use Attribute;
+use Closure;
+use Tempest\Validation\Rule;
+
+#[Attribute(Attribute::IS_REPEATABLE)]
+final readonly class Custom implements Rule
```
Comment (brendt) [inline src/Tempest/Validation/src/Rules/Custom.php:12]: Nice PR! The only thing I'm not convinced about it the name. Any ideas for something better? Maybe `Validate` ? Maybe something else?

**Review by brendt — CHANGES_REQUESTED** [review]

File: src/Tempest/Validation/src/Rules/Custom.php (hunk 1 above)
Comment (innocenzi) [reply src/Tempest/Validation/src/Rules/Custom.php:12]: Maybe `CallbackRule` or `Callback`?

Comment (Phil-Venter) [reply src/Tempest/Validation/src/Rules/Custom.php:12]: @innocenzi @brendt you apposed to `LambdaValidation` or is that too obscure??

Comment (innocenzi) [reply src/Tempest/Validation/src/Rules/Custom.php:12]: Lambda is not a term used much in PHP-land to refer to closures, so I'm personally not a fan of it 😞

Comment (Phil-Venter) [reply src/Tempest/Validation/src/Rules/Custom.php:12]: Alright, point taken, will be using `Callback` and based on the namespace it should be obvious...

**Comment by brendt** [comment]
Uh, I just had a realisation: is it even possible to pass closures to an attribute? I don't think it is? https://3v4l.org/kCoEl

**Pull request closed without merging**